src/core/helpers/ability.js

```diff
@@ -70,13 +71,16 @@ export const defineAbilitiesForCourse = (user) => {
     if ([VENDOR_ADMIN, TRAINING_ORGANISATION_MANAGER].includes(vendorRole)) {
       can('update', 'Course', 'company_representative', { type: { $in: [INTRA, INTRA_HOLDING] } });
       can('update', 'interlocutor');
+      can('read', 'companies');
```

Comment:
En fait pas certaine d'avoir bien compris les arguments qu'on passe a `can` : dans certains cas tu passes un subject en deuxieme argument et dans d'autres tu passes juste le champ concerné par la permission, pourquoi ?

Reply:
parce que le subject on en a besoin pour savoir sur quoi on base la condition (= { type: { $in: [INTER_B2B, INTRA_HOLDING] }  } par exemple) mais quand y en a pas genre read, companies ou update, interlocutor, il n'y a pas de subject à passer

Reply:
En relisant la doc, j'ai l'impression que la bonne pratique quand on fait de la programmation fonctionnelle est d'utiliser le helper subject dans tous les cas. Il permet de typer l'objet JS sur lequel on travaille :  https://casl.js.org/v6/en/guide/subject-type-detection#how-does-casl-detect-subject-type:~:text=How%20does%20CASL-,detect,-subject%20type%3F

Reply:
mais en l'occurence ici je travaille pas sur un objet, tu parles de quelle partie dans la doc ?

Reply:
Si, on travaille sur un objet : Course

Ce qui me derange dans ce que tu as ecrit c'est qu'on ne sait pas sur quel objet on regarde la permission d'accéder en lecture a 'companies'. Je me pose la question de pourquoi tu n'as pas plutot ecrit: 
```
can('read', 'Course', 'companies');
```
et dans `canAccessCompany` :
```
return ability.can('update', subject('Course', course.value), 'companies');
```
C'est une vraie question. Je ne dis pas qu'il faut faire comme je propose mais en lisant la doc j'ai l'impression qu'en programmation fonctionnelle ils preconisent de plutot utiliser les notations ci-dessus

Reply:
En comparant les 2 notations, je trouve confusant et etonnant de : 
- dans un cas passer : une action, un subject et un champ (`can('update', 'Course', 'companies');`)
- dans un autre passer: une action, un subject qui en realite est un field (`can('read', 'companies');`